detection-rules/impersonation_recipient_domain.yml

```diff
@@ -0,0 +1,35 @@
+name: "Impersonation using recipient domain (first-time sender)"
+description: |
+  The recipient's domain is used in the sender's display name
+  in order to impersonate the organization. The impersonation has been 
+  observed to use both the recipient's full email address, as well as 
+  just the domain.
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+
+  // only 1 To: recipient
+  and length(recipients.to) + length(recipients.bcc) == 1
+  and length(recipients.cc) == 0
```

Comment:
It looks like it's possible to CC someone and send with empty To and BCC lines, so we might include CC in the check on line 13?